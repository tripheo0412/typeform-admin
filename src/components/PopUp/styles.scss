@import "../../assets/styles/variables";
@import "../../assets/styles/mixins";
@import "../../assets/styles/normalize";

.modal {
  @include font-size-m;
  font-family: $font-family;
  font-weight: 300;

  display: block;
  width: 500px;
  max-width: 100%;
  position: fixed;
  z-index: 100;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  box-shadow: 0 0 10px 0px $disabled-dark;
  border-radius: 10px;
  @include color($white, $dark-light);

  &__search {
    border-radius: 0;
    &--results {
      &--panel {
        &--header {
          @include font-size-s;
          color: $disabled-dark;
          padding-left: $spacing-02;
          padding-right: $spacing-02;
          padding-bottom: $spacing-00-50;
        }
        &--item {
          cursor: pointer;
          padding-left: $spacing-02;
          padding-right: $spacing-02;
          padding-top: $spacing-00-50;
          padding-bottom: $spacing-00-50;

          &:hover {
            background: rgb(229, 243, 243);
          }
        }
      }
    }
  }

  &__wrapper {
    padding: $spacing-02;
    border-radius: 10px;

    &--dark {
      @include color($dark-light, $white);
    }
    .closed {
      display: none;
    }

    &--header {
      @include font-size-l;
    }

    &--close-btn {
      cursor: pointer;
      position: absolute;
      z-index: 1;
      top: 15px;
      right: 20px;
      background: none;
      outline: none;

      border: 0;
      padding-top: $spacing-00-25;
      font-size: 30px;

      &:hover {
        color: darkgray;
      }

      &--dark {
        color: $white;
      }
    }

    &--content {
      position: relative;
    }

    &--footer {
      &--buttons {
        display: flex;
        justify-content: flex-end;

        button {
          margin-left: $spacing-01;
          text-transform: capitalize;
        }
      }
    }
  }
}

@include for-phone-only {
  .modal {
    width: 300px;
  }
}
