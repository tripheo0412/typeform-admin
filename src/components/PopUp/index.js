// @flow
import * as React from 'react';
import classNames from 'classnames';

import InputField from '../InputField';
import Button from '../Button';

import { ThemeContext } from '../../contexts/ThemeContext';

import './styles.scss';

type ButtonSubmit = 'save' | 'delete' | 'edit' | 'ok' | 'create workspace';

type Props = {
  title?: string,
  isSearch?: boolean,
  searchResults?: string[],
  buttonSubmit?: ButtonSubmit,
  handleSubmit?: () => void,
  handleCancel?: () => void,
};

const modalInputProps = {
  id: '',
  labelText: '',
  type: 'text',
  placeholder: '',
  value: '',
  variant: 'template',
  isSignin: false,
};

export const PopUp = ({
  title,
  isSearch,
  searchResults,
  handleSubmit,
  buttonSubmit,
  handleCancel,
}: Props) => {
  const { dark } = React.useContext(ThemeContext);
  const [isClose, setIsClose] = React.useState(true);
  const [state, setState] = React.useState('');

  const modalWrapperStyles = classNames({
    modal__wrapper: true,
    'modal__wrapper--dark': dark,
  });

  const modalWrapperCloseButtonStyles = classNames({
    'modal__wrapper--close-btn': true,
    'modal__wrapper--close-btn--dark': dark,
  });
  // eslint-disable-next-line no-console
  console.log(state);

  const getVariant = value => {
    switch (value) {
      case 'ok' || 'edit' || 'save' || 'create workspace':
        return 'primary';

      case 'delete':
        return 'danger';

      case 'duplicate':
        return 'duplicate';

      default:
        return 'primary';
    }
  };

  const getLabel = value => {
    switch (value) {
      case 'ok':
        return 'ok';

      case 'edit':
        return 'edit';

      case 'save':
        return 'save';

      case 'delete':
        return 'delete';

      case 'create workspace':
        return 'create workspace';

      default:
        return '';
    }
  };

  const handleClose = () => setIsClose(!isClose);

  const handleChange = (e: { target: HTMLInputElement }) => {
    const { value } = e.target;
    setState(value);
  };

  return (
    <>
      {isSearch ? (
        <div className="modal modal__search">
          {' '}
          <div className="modal__search">
            <div className="modal__search--input">
              <InputField
                {...modalInputProps}
                variant="search"
                placeholder="Find a workspace or template"
                handleChange={handleChange}
              />
            </div>
            <div className="modal__search--results">
              {searchResults && searchResults.length > 0 && (
                <div className="modal__search--results--panel">
                  <span className="modal__search--results--panel--header">
                    WORKSPACES
                  </span>
                  {searchResults.map(s => (
                    <div className="modal__search--results--panel--item">
                      {s}
                    </div>
                  ))}
                </div>
              )}
            </div>
          </div>
        </div>
      ) : (
        isClose && (
          <div className="modal">
            <div className={modalWrapperStyles}>
              <button
                type="submit"
                className={modalWrapperCloseButtonStyles}
                onClick={handleClose}
              >
                &times;
              </button>
              <div className="modal__wrapper--header">
                <span>
                  <b>{title}</b>
                </span>
              </div>
              <br />
              <div className="modal__wrapper--content">
                <InputField
                  {...modalInputProps}
                  variant="template"
                  handleChange={handleChange}
                />
              </div>
              <br />
              <div className="modal__wrapper--footer--buttons">
                <Button
                  variant={getVariant(buttonSubmit)}
                  size="sm"
                  label={getLabel(buttonSubmit)}
                  onClick={handleSubmit}
                />
                <Button
                  variant="secondary"
                  size="sm"
                  label="Cancel"
                  onClick={handleCancel}
                />
              </div>
            </div>
          </div>
        )
      )}
    </>
  );
};

export default PopUp;
