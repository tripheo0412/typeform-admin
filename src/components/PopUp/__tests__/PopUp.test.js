import React from 'react';
import { shallow } from 'enzyme';

import PopUp from '..';

describe('popup omponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PopUp debug />);
    expect(component).toMatchSnapshot();
  });
});
