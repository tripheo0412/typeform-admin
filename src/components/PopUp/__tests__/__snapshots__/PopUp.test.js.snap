// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup omponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
